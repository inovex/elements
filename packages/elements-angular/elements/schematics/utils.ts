import { virtualFs, workspaces } from '@angular-devkit/core';
import { SchematicsException, Tree } from '@angular-devkit/schematics';

const CONFIG_PATH = 'angular.json';

function createHost(tree: Tree): workspaces.WorkspaceHost {
  return {
    async readFile(path: string): Promise<string> {
      const data = tree.read(path);
      if (!data) {
        throw new SchematicsException('File not found.');
      }
      return virtualFs.fileBufferToString(data);
    },
    async writeFile(path: string, data: string): Promise<void> {
      return tree.overwrite(path, data);
    },
    async isDirectory(path: string): Promise<boolean> {
      return !tree.exists(path) && tree.getDir(path).subfiles.length > 0;
    },
    async isFile(path: string): Promise<boolean> {
      return tree.exists(path);
    },
  };
}

export function readConfig(host: Tree) {
  const sourceText = host.read(CONFIG_PATH)?.toString('utf-8') || '';
  return JSON.parse(sourceText);
}

export function writeConfig(host: Tree, config: JSON) {
  host.overwrite(CONFIG_PATH, JSON.stringify(config, null, 2));
}

export function getAngularAppConfig(
  config: any,
  projectName: string
): any | never {
  if (!config.projects.hasOwnProperty(projectName)) {
    throw new SchematicsException(`Could not find project: ${projectName}`);
  }

  return config.projects[projectName];
}

export async function getProjectName(tree: Tree) {
  const host = createHost(tree);
  const { workspace } = await workspaces.readWorkspace('/', host);
  return workspace.extensions.defaultProject as string;
}
