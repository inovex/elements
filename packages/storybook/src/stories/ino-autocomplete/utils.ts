import { KeyValue } from '@inovex.de/elements';

export const AutoCompleteStoryOptions: KeyValue[] = [
  {
    key: 'hh',
    value: 'Hamburg',
  },
  {
    key: 'ber',
    value: 'Berlin',
  },
  {
    key: 'k',
    value: 'Karlsruhe',
  },
];
