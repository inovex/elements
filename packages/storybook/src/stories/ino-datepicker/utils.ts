import '@inovex.de/elements';

export function registerInlineDatepickerHandler() {
  const group = document.querySelector('ino-radio-group') as HTMLInoRadioGroupElement;
  const datepicker = document.querySelector('.datepicker-group ino-datepicker') as HTMLInoDatepickerElement;

  const today = document.querySelector('.today');
  const thisWeek = document.querySelector('.thisWeek');
  const lastWeek = document.querySelector('.lastWeek');

  const todayClickHandler = () => {
    datepicker.value = '02.03.2020';
    datepicker.label = 'Am';
    group.value = 'at';
    datepicker.range = false;
  };

  const thisWeekClickHandler = () => {
    group.value = 'range';
    datepicker.label = 'Zeitraum';
    datepicker.range = true;
    datepicker.value = '02.03.2020 bis 08.03.2020';
  };

  const lastWeekClickHandler = () => {
    group.value = 'range';
    datepicker.label = 'Zeitraum';
    datepicker.range = true;
    datepicker.value = '24.02.2020 bis 01.03.2020';
  };

  const checkedChangeHandler = (e: CustomEvent<boolean>) => {
    const target = e.target as HTMLInoRadioElement;

    switch (target.value) {
      case 'at':
        datepicker.label = 'Am';
        datepicker.range = false;
        break;
      case 'after':
        datepicker.label = 'Ab';
        datepicker.range = false;
        if (group.value == 'range') {
          datepicker.value = datepicker.value.split(' - ').shift();
        }
        break;
      case 'before':
        datepicker.label = 'Bis';
        datepicker.range = false;
        if (group.value == 'range') {
          datepicker.value = datepicker.value.split(' - ').pop();
        }
        break;
      case 'range':
        datepicker.range = true;
        datepicker.label = 'Zeitraum';
        break;
    }
    group.value = target.value;
  };

  today.addEventListener('click', todayClickHandler);
  thisWeek.addEventListener('click', thisWeekClickHandler);
  lastWeek.addEventListener('click', lastWeekClickHandler);
  group.addEventListener('checkedChange', checkedChangeHandler);

  return () => {
    today.removeEventListener('click', todayClickHandler);
    thisWeek.removeEventListener('click', thisWeekClickHandler);
    lastWeek.removeEventListener('click', lastWeekClickHandler);
    group.removeEventListener('checkedChange', checkedChangeHandler);
  };
}
