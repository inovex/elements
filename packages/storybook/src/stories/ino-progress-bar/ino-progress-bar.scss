.story-ino-progress-bar {
  .progress-bar-example {
    display: flex;
    flex-direction: row;
    width: 100%;
    justify-content: space-around;
    align-items: center;

    ino-progress-bar {
      width: 80%;
    }

    ino-input-file {
      width: 10%;
    }
  }
}
