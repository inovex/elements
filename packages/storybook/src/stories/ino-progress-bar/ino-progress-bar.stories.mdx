import { Meta, Description } from "@storybook/addon-docs";
import readme from "../../../../elements/src/components/ino-progress-bar/readme.md";

<Meta title="Notification/ino-progress-bar/Documentation" />

<Description markdown={readme} />
