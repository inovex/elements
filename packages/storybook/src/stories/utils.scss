/**
 * Applies styling to the container of a story.
 * The styling is encapsulated within this mixin because the ID may change in the future.
 */
@mixin story-container {
  #root-inner {
    @content;
  }
}
