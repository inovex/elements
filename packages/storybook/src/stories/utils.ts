import { html } from 'lit-html';
import ICONS from '../../../elements/src/components/ino-icon/icons';

export const maybeCreateStoryArgs = (story: any) => {
  if (!story.args) {
    story.args = {};
  }
  if (!story.argTypes) {
    story.argTypes = {};
  }
};

export const withIconControl = (
  story: any,
  propertyName: string,
  defaultValue?: string
) => {
  maybeCreateStoryArgs(story);
  story.args[propertyName] = defaultValue || '';
  story.argTypes[propertyName] = {
    control: {
      type: 'select',
    },
    options: getIcons(),
  };
};

export const withColorScheme = (
  story: any,
  propertyName: string,
  defaultValue?: string
) => {
  maybeCreateStoryArgs(story);
  story.args[propertyName] = defaultValue || '';
  story.argTypes[propertyName] = {
    control: {
      type: 'select',
    },
    options: [
      'primary',
      'secondary',
      'success',
      'warning',
      'error',
      'light',
      'dark',
    ],
  };
};

export const withSortDirection = (
  story: any,
  propertyName: string,
  defaultValue?: string
) => {
  maybeCreateStoryArgs(story);
  story.args[propertyName] = defaultValue || '';
  story.argTypes[propertyName] = {
    options: ['desc', 'asc'],
    control: {
      type: 'select',
    },
  };
};

export const getIcons = () => ICONS.filter((icon) => !icon.startsWith('_'));

export const defaultDecorator = (story, className?: string) => {
  return html` <div class="${className ?? ''}">${story()}</div> `;
};
