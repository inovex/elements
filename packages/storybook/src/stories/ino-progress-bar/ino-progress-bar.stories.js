import componentReadme from '_local-elements/src/components/ino-progress-bar/readme.md';
import withStencilReadme from '_local-storybookcore/with-stencil-readme';
import './ino-progress-bar.scss';

function subscribeToComponentEvents() {
  // == event block

  const handleFileSelection = (e) => {
    const bar = document.querySelector('#example');
    let totalFileSize = 0;
    let loaded = 0;
    const files = e.detail.files;

    if (files) {
      for (let i = 0; i < files.length; i++) {
        totalFileSize += files[i].size;
      }
      for (let i = 0; i < files.length; i++) {
        let reader = new FileReader();
        reader.onloadend = (e) => {
          loaded += e.total;
        };
        reader.onprogress = (e) => {
          bar.inoProgress = (loaded + e.loaded) / totalFileSize;
        };
        reader.readAsDataURL(files[i]);
      }
    }
  };

  const handleClick = (e) => {
    if (e.target.tagName.toLowerCase() === 'ino-button') {
      const bar = document.querySelector('#example');
      bar.inoProgress = 0;
    }
  };

  document.addEventListener('changeFile', handleFileSelection);
  document.addEventListener('click', handleClick);
  // == event block

  // unsubscribe function will be called by Storybook
  return () => {
    document.removeEventListener('changeFile', handleFileSelection);
    document.removeEventListener('click', handleClick);
  };
}

export default {
  title: 'Notification/<ino-progress-bar>',

  decorators: [
    withStencilReadme(componentReadme),
    (story) => {
      subscribeToComponentEvents();
      return story();
    },
  ],
};

export const DefaultUsage = () => /* html */ `

    <h4>With custom max and min values</h4>
    <ino-progress-bar ino-progress="9"></ino-progress-bar>


    <h3>Example</h3>
    <div class="progress-bar-example">
        <ino-progress-bar ino-progress="0" ino-buffer="1" id="example"></ino-progress-bar>
        <ino-input-file ino-label="Upload" multiple></ino-input-file>
    </div>


    <h3>Variants</h3>
    <h4>With buffering dots</h4>
    <ino-progress-bar ino-buffer="0.8" ino-progress="0.3"></ino-progress-bar>

    <h4>Without buffering dots</h4>
    <ino-progress-bar ino-buffer="1" ino-progress="0.3"></ino-progress-bar>

    <h4>Indeterminate progress bar</h4>
    <ino-progress-bar ino-indeterminate></ino-progress-bar>

    <h4>Inverted progress bar</h4>
    <ino-progress-bar ino-reversed ino-progress="0.6" ino-buffer="0.8"></ino-progress-bar>
    `;
