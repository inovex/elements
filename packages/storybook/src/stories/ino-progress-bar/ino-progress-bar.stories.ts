import { Components } from '@inovex.de/elements';
import { html } from 'lit-html';
import { defaultDecorator } from '../utils';
import './ino-progress-bar.scss';
import { useEffect } from '@storybook/client-api';
import { Story } from '@storybook/web-components';

export default {
  title: 'Notification/ino-progress-bar',
  component: 'ino-progress-bar',
  decorators: [
    s => defaultDecorator(s, 'story-ino-progress-bar'),
    story => {
      const clickHandler = (e) => {
        if (e.target.tagName.toLowerCase() === 'ino-button') {
          const bar = document.querySelector('#example') as HTMLInoProgressBarElement;
          bar.progress = 0;
        }
      };

      const changeFileHandler = e => {
        console.log("changeFileHandler");
        const bar = document.querySelector('#example') as HTMLInoProgressBarElement;
        let totalFileSize = 0;
        let loaded = 0;
        const files = e.detail.files;

        if (files) {
          for (let i = 0; i < files.length; i++) {
            totalFileSize += files[i].size;
          }
          for (let i = 0; i < files.length; i++) {
            let reader = new FileReader();
            reader.onloadend = (e) => {
              loaded += e.total;
            };
            reader.onprogress = (e) => {
              bar.progress = (loaded + e.loaded) / totalFileSize;
            };
            reader.readAsDataURL(files[i]);
          }
        }
      };
      useEffect(() => {
        document.addEventListener('click', clickHandler);
        document.addEventListener('changeFile', changeFileHandler);

        return () => {
          document.removeEventListener('click', clickHandler);
          document.removeEventListener('changeFile', changeFileHandler);
        }
      });
      return story();
    }],
};

export const Playground: Story<Components.InoProgressBar>= (args) => {
  return html`
    <ino-progress-bar
      buffer="${args.buffer}"
      progress="${args.progress}"
      reversed="${args.reversed}"
      indeterminate="${args.indeterminate}"
      label="${args.label}"
    >
    </ino-progress-bar>
  `;
}
Playground.args = {
  buffer: 0.9,
  progress: 0.4,
  reversed: false,
  indeterminate: false,
  label: 'Progress Bar',
}
Playground.argTypes = {
  buffer: {
    control: { min: 0, max: 1, step: 0.1 }
  },
  progress: {
    control: { min: 0, max: 1, step: 0.1 }
  }
}

export const Variants = () => html`
  <h5>With buffering dots</h5>
  <ino-progress-bar buffer="0.8" progress="0.3"></ino-progress-bar>

  <h5>Without buffering dots</h5>
  <ino-progress-bar buffer="1" progress="0.3"></ino-progress-bar>

  <h5>Indeterminate progress bar</h5>
  <ino-progress-bar indeterminate></ino-progress-bar>

  <h5>Reversed progress bar</h5>
  <ino-progress-bar reversed progress="0.6" buffer="0.8"></ino-progress-bar>
`;

export const UploadExample = () => html`
  <div class="progress-bar-example">
    <ino-progress-bar
      id="example"
      progress="0"
      buffer="1"
    >
    </ino-progress-bar>
    <ino-input-file label="Upload" multiple></ino-input-file>
  </div>
`;
