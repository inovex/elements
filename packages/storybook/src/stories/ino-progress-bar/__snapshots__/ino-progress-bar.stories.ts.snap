// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Notification/ino-progress-bar Buffer smoke-test 1`] = `
<div id="root-inner">
  <div class="story-ino-progress-bar">
    <ino-progress-bar buffer="0.5"
                      progress="undefined"
                      indeterminate="false"
                      class="hydrated"
    >
    </ino-progress-bar>
  </div>
</div>
`;

exports[`Notification/ino-progress-bar Indeterminate smoke-test 1`] = `
<div id="root-inner">
  <div class="story-ino-progress-bar">
    <ino-progress-bar buffer="undefined"
                      progress="undefined"
                      indeterminate="true"
                      class="hydrated"
    >
    </ino-progress-bar>
  </div>
</div>
`;

exports[`Notification/ino-progress-bar Playground smoke-test 1`] = `
<div id="root-inner">
  <div class="story-ino-progress-bar">
    <ino-progress-bar buffer="0.9"
                      progress="0.4"
                      indeterminate="false"
    >
    </ino-progress-bar>
  </div>
</div>
`;

exports[`Notification/ino-progress-bar Progress smoke-test 1`] = `
<div id="root-inner">
  <div class="story-ino-progress-bar">
    <ino-progress-bar buffer="undefined"
                      progress="0.5"
                      indeterminate="false"
                      class="hydrated"
    >
    </ino-progress-bar>
  </div>
</div>
`;
